pkg/deploy/util/util.go

```diff
@@ -72,6 +73,13 @@ func GetDeploymentCondition(status deployapi.DeploymentConfigStatus, condType de
 
 // SetDeploymentCondition updates the deployment to include the provided condition.
 func SetDeploymentCondition(status *deployapi.DeploymentConfigStatus, condition deployapi.DeploymentCondition) {
+	// If the condition that we are about to add already exists and has the same status
+	// then all we need is to update its LastProbe time.
+	currentCond := GetDeploymentCondition(*status, condition.Type)
+	if currentCond != nil && currentCond.Status == condition.Status && currentCond.Reason == condition.Reason {
```

Comment:
Doesn't this mean we are updating all deployments frequently?  How many extra writes does this add per deployment?  How frequently will we write now (since all condition changes now result in writes to etcd)?